junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/TestExecutionListenerRegistry.java

```diff
@@ -63,48 +67,117 @@ private void notifyEagerTestExecutionListeners(Consumer<EagerTestExecutionListen
 		this.eagerTestExecutionListeners.forEach(consumer);
 	}
 
-	TestExecutionListener getCompositeTestExecutionListener() {
+	CompositeTestExecutionListener getCompositeTestExecutionListener() {
 		return new CompositeTestExecutionListener();
 	}
 
-	private class CompositeTestExecutionListener implements TestExecutionListener {
+	class CompositeTestExecutionListener implements TestExecutionListener {
+		public List<TestExecutionListener> getAll() {
+			return testExecutionListeners;
+		}
 
 		@Override
 		public void dynamicTestRegistered(TestIdentifier testIdentifier) {
-			notifyTestExecutionListeners(listener -> listener.dynamicTestRegistered(testIdentifier));
+			notifyTestExecutionListeners(listener -> {
+				try {
+					listener.dynamicTestRegistered(testIdentifier);
+				}
+				catch (Throwable throwable) {
+					rethrowIfBlacklistedAndLogTest(throwable, listener, testIdentifier, "dynamicTestRegistered");
+				}
```

Comment:
Please move the exception handling into `notifyTestExecutionListeners()` instead of duplicating it.

Reply:
I thought of doing the same, but we will not get the exact listener method name on logging it in `notifyTestExecutionListeners()`.

Reply:
Why don't you just pass the method name to `notifyTestExecutionListeners()` as an additional parameter?

Reply:
Yes, we can, 
There are two variables here,
1. method name 
2. the log for `plan` level listener method needs `TestPlan`  & test level listener method need `TestIdentifier`.


Them, we need two variants of `notifyTestExecutionListeners()` one taking `TestPlan` as parameter & second taking `TestIdentifier`.

how does that sounds to you @sbrannen 




Thanks,
Gopi

Reply:
I think that would be reasonable and still less code duplication. Give it a shot.

Reply:
Sure. Let's give it a shot and see 👍

Reply:
@sbrannen  @marcphilipp 

I was thinking to extract `TestExecutionListenerNotifier` class out to extract the notification logic at one place from `TestExecutionListenerRegistry`, `ExecutionListenerAdapter` or (may be from `DefaultLauncher`.

Currently, this logic (handling with try/catch & logging is at multiple class.


The resulting class will look like, 

```
public class TestExecutionListenerNotifier {
	private static final Logger logger = LoggerFactory.getLogger(TestExecutionListenerNotifier.class);

	public static void notify(TestExecutionListener listener, TestIdentifier testIdentifier, String listenerMethodName, Consumer<TestExecutionListener> consumer) {
		try {
			consumer.accept(listener);
		}
		catch (Throwable throwable) {
			rethrowIfBlacklistedAndLogTest(throwable, listener, testIdentifier, listenerMethodName);
		}

	}
	private static void rethrowIfBlacklistedAndLogTest(Throwable throwable, TestExecutionListener listener,
												TestIdentifier testIdentifier, String methodName) {
		BlacklistedExceptions.rethrowIfBlacklisted(throwable);
		logger.warn(throwable,
				() -> String.format("Failed to invoke ExecutionListener [%s] for method [%s] with test display name [%s]",
						listener.getClass().getName(), methodName, testIdentifier.getDisplayName()));
	}

}```

Here, I am thinking to use static methods.

Any thought on this?

Reply:
We don't need to handle exceptions in `ExecutionListenerAdapter` because it's an internal class that's only ever called with a `CompositeTestExecutionListener` which now handles exceptions.

Reply:
Sure. I will go through it one more time.

thanks for feedback I will get back with commits :)